proposals/032-deprecate-unintended-apis.md

```diff
@@ -0,0 +1,221 @@
+
+* For COM interfaces:
+
+    * if no other interfaces in the public API inherit from them, mark them ``[[deprecated]]`
+
+    * Otherwise, mark all of their methods as `[[deprecated]]` and add a commend indicating that the entire interface is deprecated
+
+* In cases where an interface method should be *renamed* (with no signature change) as opposed to just deprecated:
+
+    * Change the name of the `virtual` method
+
+    * Introduce a `[[deprecated]]` inline method with the old name that delegates to the `virtual` one
+
+* If a symbol can be moved into `slang-deprecated.h` without breaking other declarations in `slang.h`, it should be moved there
+
+The features we propose to deprecate are:
+
+* interface `ISlangCastable`
+
+* interface `ISlangClonable`
+
+* `struct SlangTerminatedChars`
+
+* type `SlangPathTypeIntegral`
+
+* `enum SlangPathType`
+
+* type `FileSystemContentsCallBack`
+
+* `enum OSPathKind`
+
+* `enum PathKind`
+
+* interface `ISlangFileSystemExt`
+
+* interface `ISlangMutableFileSystem`
+
+* type `SlangWriterChannelIntegral`
+
+* `enum SlangWriterChannel`
+
+* type `SlangWriterModeIntegral`
+
+* `enum SlangWriterMode`
+
+* interface `ISlangWriter`
+
+* interface `ISlangProfiler`
+
+* type `SlangDiagnosticCallback`
+
+* method `slang::IGlobalSession::addBuiltins`
+
+* method `slang::IGlobalSession::setDownstreamCompilerPrelude`
+
+* method `slang::IGlobalSession::getDownstreamCompilerPrelude`
+
+* method `slang::IGlobalSession::setLanguagePrelude`
+
+* method `slang::IGlobalSession::getLanguagePrelude`
+
+* method `slang::IGlobalSession::checkPassThroughSupport`
+
+* method `slang::IGlobalSession::compileCoreModule`
+
+* method `slang::IGlobalSession::loadCoreModule`
+
+* method `slang::IGlobalSession::saveCoreModule`
+
+* method `slang::IGlobalSession::saveCoreModule`
+
+* method `slang::IGlobalSession::getCompilerElapsedTime`
+
+* method `slang::IGlobalSession::compileBuiltinModule`
+
+* method `slang::IGlobalSession::loadBuiltinModule`
+
+* method `slang::IGlobalSession::saveBuiltinModule`
+
+* `enum slang::BuiltinModuleName`
+
+* `enum SlangArchiveType`
+
+* type `CompileCoreModuleFlags`
+
+* `struct CompileCoreModuleFlag`
+
+* method `slang::ISession::createCompileRequest`
+
+* method `slang::ISession::loadModuleFromIRBlob` should be renamed to `loadModuleFromBinaryBlob`
+
+* method `slang::IComponentType::getResultAsFileSystem`
+
+* method `slang::IComponentType::getEntryPointHostCallable`
+
+* function `slang_createGlobalSessionWithoutCoreModule`
+
+* function `slang_getEmbeddedCoreModule`
+
+* option `-emit-ir` should be deprecated
+
+* option `-report-downstream-time` should be marked as internal
+
+* option `-report-perf-benchmark` should be marked as internal
+
+* option `-report-checkpoint-intermediates` should be marked as internal
+
+* option `-source-embed-style` should be deprecated
+
+* option `-source-embed-name` should be deprecated
+
```

Comment:
I am actively using these options, they're useful for small applications that want to embed SPIR-V in the executable so that they don't need to care about a filesystem. They also mirror equivalent `glslc` / `glslang` functionality, making `slangc` a drop-in replacement in an existing CMake shader compilation setup.

I have no issues with the other deprecations listed.

Reply:
I guess I'm also not the only person using this functionality, here's someone recently on Discord asking how to do exactly what these options do: https://discord.com/channels/1303735196696445038/1397550571321299006